test/abstract_syntax_tree.coffee

```diff
@@ -261,34 +710,246 @@ test "AST as expected for Elision node", ->
+    type: 'While'
+    condition:
+      type: 'BooleanLiteral'
+      value: 'true'
+      originalValue: 'true'   # TODO: This should probably be changed for Prettier.
+    body: type: 'Value'
+
+  testExpression 'while 1 < 2 then',
+    type: 'While'
+    condition: type: 'Op'
+    body: type: 'Block'
 
 test "AST as expected for Op node", ->
-  # TODO
+  testExpression '1 <= 2',
+    type: 'Op'
+    operator: '<='
+    originalOperator: '<='
+    flip: no
+    first: value: '1'
+    second: value: '2'
+
+  testExpression '1 is 2',
+    type: 'Op'
+    operator: '==='
+    originalOperator: 'is'
+    flip: no
+
+  testExpression '1 // 2',
+    type: 'Op'
+    operator: '//'
+    originalOperator: '//'
+    flip: no
+
+  testExpression '1 << 2',
+    type: 'Op'
+    operator: '<<'
+    originalOperator: '<<'
+    flip: no
+
+  testExpression 'new Old',   # NOTE: `new` with params is a `Call` node.
+    type: 'Op'
+    operator: 'new'
+    originalOperator: 'new'
+    flip: no
+    first: value: 'Old'
 
 test "AST as expected for In node", ->
-  # TODO
+  testExpression '1 in 2',
+    type: 'Op'
+    operator: 'in'
+    originalOperator: 'in'
+    flip: no
+    first: value: '1'
+    second: value: '2'
+
+  testExpression 'for x in 2 then',
+    type: 'For'
+    range: no
+    pattern: no
 
 test "AST as expected for Try node", ->
-  # TODO
+  testExpression 'try cappuccino',
+    type: 'Try'
+    attempt: type: 'Value'
+    recovery: undefined
+
+  testExpression 'try to catch it then log it',
+    type: 'Try'
+    attempt: type: 'Value'
+    recovery:
+      type: 'Value'
+      base: type: 'Call'
 
 test "AST as expected for Throw node", ->
-  # TODO
+  testExpression 'throw new BallError "catch"',
+    type: 'Throw'
+    expression:
+      type: 'Call'
+      isNew: yes
 
 test "AST as expected for Existence node", ->
-  # TODO
+  testExpression 'Ghosts?',
+    type: 'Existence',
+    comparisonTarget: 'null'
+    expression: value: 'Ghosts'
+
+  # NOTE: Soaking is covered in `Call` and `Access` nodes.
 
 test "AST as expected for Parens node", ->
-  # TODO
+  testExpression '(hmmmmm)',
+    type: 'Parens',
+    body: type: 'Value'
+
+  testExpression '(a + b) / c',
+    type: 'Op'
+    operator: '/'
+    first:
+      type: 'Parens'
+      body:
+        type: 'Op'
+        operator: '+'
+
+  testExpression '(((1)))',
+    type: 'Parens',
+    body:
+      type: 'Value'
+      base:
+        type: 'Block'
+        expressions: [
+          type: 'Parens',
+          body:
+            type: 'Value'
+            base: value: '1'
+        ]
 
 test "AST as expected for StringWithInterpolations node", ->
-  # TODO
+  testExpression '"#{o}/"',
+    type: 'StringWithInterpolations'
+    quote: '"'
+    body:
+      type: 'Block'
+      expressions: [
+        originalValue: ''
+      ,
+        type: 'Interpolation'
+        expression:
+          type: 'Value'
+          base: value: 'o'
+      ,
+        originalValue: '/'
+      ]
```

Comment:
For strings, we will want to test (eventually) that the AST outputs the string delimiter (single quote or double quote). I assume that’s not output yet, so maybe add a TODO?

Reply:
That information is available in the `quote` property and is even checked here. I added the check to `StringLiteral` now.

Reply:
Ah, sorry I didn’t notice. Good catch.